ADFSToolkit/Public/get-ADFSTkFederationDefaults.ps1

```diff
@@ -0,0 +1,168 @@
+if ( ($URL -eq $null) -and !(Test-Path ( $federationConfigCacheFile )  ) )
+   
+    {
+        
+            Write-Output ("ADFSToolkit: Federation default behaviour expected with no extra federation settings.")
+        
+            if ($InstallDefaults)
+            {
+             Write-Output ("ADFSToolkit: InstallDefaults flag  found  but no defaults to apply, nothing changed, exiting.")
+            }
+
+
+            return
+        }
+# Use case B: no url, config cache detected, tell user about it
+elseif ( ($URL -eq $null) -and (Test-Path ( $federationConfigCacheFile )  ) )
+     {
+        Write-Output ("ADFSToolkit: Federation defaults in the cache are:")
+
+        Get-ChildItem $federationConfigCacheFile -Force | Select-Object FullName, CreationTime, LastAccessTime, LastWriteTime, Mode, Length
+
+    }
+#Use case C:  URL is not null, and  we have a file. If InstallDefaults specified make backup and deploy.
+#
+# Note that we  are  now using  if statements for presence as the defaults could be on disk if fetched  but not installed
+# 
+
+if ( !($URL -eq $null)  )
+{
+
+    Write-Output ("ADFSToolkit: Updating federation defaults on disk from: $URL")
+
+    # make backup
+    If (Test-Path ( $federationConfigCacheFile ) )
+    {
+        Write-Output ("ADFSToolkit: Backing up $fedConfigCacheFile to $fedConfigCacheFileBackup")
+        Copy-item $federationConfigCacheFile -Destination $federationConfigCacheFileBackup
+    }
+
+    # Begin fetch file  
+    $start_time = Get-Date
+
+    Write-Output ("ADFSToolkit: Fetching $URL to $federationConfigCacheFile")
+    (New-Object System.Net.WebClient).DownloadFile($URL, $federationConfigCacheFile)
+
+    Write-Output "ADFSToolkit: Time taken: $((Get-Date).Subtract($start_time).Seconds) second(s)"
+
+    Get-ChildItem $federationConfigCacheFile -Force | Select-Object FullName, CreationTime, LastAccessTime, LastWriteTime, Mode, Length
+
+    #  interrogate the file we just fetched
+    Write-Output ("ADFSToolkit: contents of Federation Cache file: $federationConfigCacheFile :")
+
+    
+
+    $zip = [io.compression.zipfile]::OpenRead($federationConfigCacheFile)
+    
+    $zip.Entries|where-object {$_.FullName -match $ExtractionFilter }|Select-object FullName |Select -ExpandProperty "FullName"
+    $zip.Dispose()
+
+}
+
+    # deploy configuration that's on disk
+    if ($InstallDefaults -and (Test-Path ( $federationConfigCacheFile ) )  )
+    {
+
+        Write-Output ("ADFSToolkit: Installing federation defaults from: $federationConfigCacheFile ...")
+    
+        $zip = [io.compression.zipfile]::OpenRead($federationConfigCacheFile)
+        $unzippedDirName=($zip.Entries[0]).FullName
+        $unzippedFullNameDir=Join-Path (Join-path $Global:ADFSTkPaths.cacheDir $unzippedDirName) $ExtractionFilter
+
+        #backup previous federation-settings
+        move-item  -Path $Global:ADFSTkPaths.federationDir -Destination $federationConfigDirFullBackupPath
```

Comment:
This will move everything including the federation cache.
Consider this code: `Get-ChildItem $Global:ADFSTkPaths.federationDir -Exclude backup,federations.xml | Move-Item $_.fullName $Global:ADFSTkPaths.federationBackupDir`